R/juvenile_month_dynamic.R

```diff
@@ -0,0 +1,419 @@
+#' Juvenile Month Dynamics
+#' @param hypothesis
+#' @param fish list object containing tracking matrices for juveniles, juveniles at chips, and adults in ocean
+#' @param year
+#' @param month
+#' @param rearing_survival
+#' @param migratory_survival
+#' @param habitat
+#' @param ..params   
+#' @examples 
+#' fish = list(juveniles = juveniles, north_delta_fish = north_delta_fish, south_delta_fish = south_delta_fish,
+#' juveniles_at_chipps = juveniles_at_chipps, 
+#' adults_in_ocean = adults_in_ocean)
+juvenile_month_dynamic <- function(hypothesis, fish, year, month, rearing_survival, 
+                                   migratory_survival, habitat, ..params, 
+                                   avg_ocean_transition_month, stochastic) {
+  
+  juveniles <- fish$juveniles
+  lower_mid_sac_fish <- fish$lower_mid_sac_fish
+  lower_sac_fish <- fish$lower_sac_fish
+  upper_mid_sac_fish <- fish$upper_mid_sac_fish
+  sutter_fish <- fish$sutter_fish
+  yolo_fish <- fish$yolo_fish
+  san_joaquin_fish <- fish$san_joaquin_fish
+  north_delta_fish <- fish$north_delta_fish
+  south_delta_fish <- fish$south_delta_fish
+  juveniles_at_chipps <- fish$juveniles_at_chipps
+  adults_in_ocean <- fish$adults_in_ocean
+  
+  if (hypothesis == 3) {
+    route <- lateFallRunDSM::route_alternative
```

Comment:
This may not work because route_alternative and the two new route functions below need the temperature downstream as an input. Maybe the easy solution is to add temp downstream as an input for the original route functions, but don't use them in the actual function. I inserted which downstream temps to use in my last code submission. Let me know if I need to set that up again

Reply:
Also, talkin with Jim we decided having a temp barrier when fish are moving to the delta (both south and north) doesn't make sense.